README.md

```diff
@@ -1,3 +1,284 @@
+
+### The `#update` Method
+
+How will we write a method that will allow us to update any attributes of any song? How will we know *which* attributes have been recently updated and which will remain the same? 
+
+The best way for us to do this is to simply update *all* the attributes whenever we update a record. That way, we will catch any changed attributes, while the un-changed ones will simply remain the same. 
+
+For example:
+
+```ruby
+hello = Song.find_by_name("Hella")
+
+sql = "UPDATE songs SET name = 'Hello', album = ? WHERE name = ?"
+
+DB[:conn].execute(sql, hello.album, hello.name)
+
+```
+
+Here we update *both* the name and album attribute of the song, even though only the album attribute is actually different. 
+
+Okay, now that we've solved this problem, let's build our method:
+
+```ruby
+class Song
+  ...
+  
+  def udpate
+    sql = "UPDATE songs SET name = ?, album = ? WHERE name = ?"
+    DB[:conn].execute(sql, self.name, self.album, self.name)
+  end
+end
+```
+
+Now we can update a song with the following:
+
+```ruby
+hello = Song.create(name: "Hella", album: "25")
+hello.name = "Hello"
+hello.update
+```
+
+## Duplication and Object ID
+
+What happens when we create another song, also called `"Hello"`? After all, this is not *such* an uncommon song title. 
+
+```ruby
+another_hello = Song.create(name: "Hello", album: "Hello!!")
+
+another_hello.album = "Hello"
+
+another_hello.update
+```
+Remember that the SQL statement inside our `#update` method looks like this:
+
+```sql
+UPDATE 
+songs SET name = ?, album = ? 
+WHERE name = ?
+```
+
+We are identifying the record we want to update by matching the *name* attribute of the object to the name value of the table row. But, song names are not unique! The above call to `#update` would update the album name of *both songs with the name of `"Hello"`. Oh no!
+
+Looks like we need to find some unique way to identify song objects and match them with their equally unique equivalent song records...
+
+Luckily for us, each song record in the songs table already has a unique value in the `id` column. That is the great thing about the primary key column––it is always unique. 
+
+Song records in the database table have a unique `id`, and our `Song` instances have an `id` attribute, but that attribute gets set to `nil` when an individual song is initialized. We want our `Song` instances to get assigned a unique `id` number, but when? and how?
+
+The unique `id` number of a `Song` instance should *come from the database*. When a song record gets inserted into the database, it automatically gets assigned a unique ID number. We need to grab that ID number *from the database record* and assign it to the `Song` instance's `id` attribute. 
+
+If that sounds confusing, check out this diagram:
+
+![](https://docs.google.com/drawings/d/1KYjce3m-ItD3j0hTuorBtlTgPoQSVg9v2sddGLVle8M/pub?w=958&h=497)
```

Comment:
@jmburges images on s3 not showing up. not sure why. It is in the s3 bucket, permission are "read: world".